dotnet-desktop-guide/net/winforms/controls-design/usercontrol-overview.md

```diff
@@ -0,0 +1,176 @@
+
+01. In the **Name** box, type a name for your user control. Visual Studio provides a default and unique name that you may use. Next, press **Add**.
+
+    :::image type="content" source="media/usercontrol-overview/new-usercontrol-dialog.png" alt-text="Add item dialog in Visual Studio for Windows Forms":::
+
+After the user control is created, Visual Studio opens the designer:
+
+:::image type="content" source="media/usercontrol-overview/designer.png" alt-text="The user control designer in Visual Studio for Windows Forms":::
+
+## Example: Create a clearable text box
+
+In the following example, you'll learn how to create a reusable user control that's both visually appealing and functional. The user control is made up of three controls: a `TextBox`, a `Label`, and a `Button`. The `TextBox` is used for entering text, while the `Button` clears it.
+
+01. Create a new Windows Forms project.
+01. [Add a user control to your project](#add-a-user-control-to-a-project). The default name of `UserControl1` is fine.
+01. The **UserControl1** designer is shown, and the user control should be the focused object. If it's not, click on the design surface to select it. Set the following properties:
+
+    | Property    | Value      |
+    |-------------|------------|
+    | MinimumSize | `84, 53`   |
+    | Size        | `191, 53`  |
+
+01. Add a **Label** control. Set the following properties:
+
+    | Property  | Value     |
+    |----------|------------|
+    | Name     | `lblTitle` |
+    | Location | `3, 5`     |
+
+01. Add a **TextBox** control. Set the following properties:
+
+    | Property  | Value     |
+    |----------|------------|
+    | Name     | `txtValue` |
+    | Anchor   | `Top, Left, Right` |
+    | Location | `3, 23`    |
+    | Size     | `148, 23`  |
+
+01. Add a **Button** control. Set the following properties:
+
+    | Property  | Value     |
+    |----------|------------|
+    | Name     | `btnClear` |
+    | Anchor   | `Top, Right` |
+    | Location | `157, 23`  |
+    | Size     | `31, 23`   |
+    | Text     | `↻`        |
+
+    The control should look like the following image:
+
+    :::image type="content" source="./media/usercontrol-overview/example-initial-design.png" alt-text="Visual Studio with Windows Forms, showing the user control that was just designed.":::
+
+01. Press <kbd>F7</kbd> to open the code editor for the `UserControl1` class.
+01. Make the following code changes:
+
+    01. Before the `class` declaration, add import the `System.ComponentModel` namespace, and add the `DefaultEvent` attribute to the class. This attribute sets which event is generated by the consumer when the control is double-clicked in the designer.
+
+        :::code language="csharp" source="./snippets/usercontrol-overview/csharp/UserControl1.cs" id="default_event":::
+
+    01. Add an event handler that exposes the [`TextBox.TextChanged`](xref:System.Windows.Forms.Control.TextChanged) event to the consumer:
```

Comment:
We could change it around. However, it's presenting a **new** textbox control in the sense of adding the clearable button. Most likely when someone creates a new textbox control, they'll want that text changed event forwarded. So it makes sense in that aspect.

Do you have an idea for a different control example? The thing is that the base class defines tons of events that are used by the various controls themselves. Whenever you create a user control with all sorts of controls added, it's probably a common scenario of needing to forward an event from the constituent control to the user control, and since the user control seems to define every possible event (lol) you're going to run into collision.

Reply:
Yes, I think what I wrote above might make sense. Like when you fill out a form, or a chat message or what have you, you could create a text length threshold property, and if that length is exceeded, you could fire that event, and then update a label on top of that TextBox.

Reply:
Let me know if I can help you with that!

Reply:
Let's do it like we discussed it in our 1:1.
Resolving this.